p2p/src/net/mock/peer.rs

```diff
@@ -90,25 +96,23 @@ where
     async fn handshake(&mut self) -> crate::Result<()> {
         match self.role {
             Role::Inbound => {
-                let (peer_id, network, version, protocols) =
-                    if let Ok(Some(types::Message::Handshake(types::HandshakeMessage::Hello {
+                let Ok(Some(types::Message::Handshake(types::HandshakeMessage::Hello {
                         peer_id,
                         version,
                         network,
                         protocols,
+                        subscriptions,
```

Comment:
Please make sure there are protections against having huge values in these Vec and BTreeSet

Reply:
I have rechecked and the subscriptions are only stored in `BTreeSet`, therefore it is impossible to send duplicated topics. Personally I would be more concerned with the `PeerInfo::agent` field because an arbitrary long string can be stored there.

Additionally, we already have the `MAX_MESSAGE_SIZE` constant, though it is equal to 10 megabytes. Do we need additional limitation specifically for subscriptions or peer info?

Reply:
Theoretically speaking, it should be enough. Though every layer should have its own protection. If you wish, you could make this an issue with testnet milestone. No hurry on this.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/594